openfisca_core/types/_domain.py

```diff
@@ -1,3 +1,20 @@
+"""Domain-model.
+
+The domain-model is composed of structures meant to encapsulate data in a way
+that is unique to the context. Therefore, their identity is not equivalent to
+the sum of their properties. If two data objects hold the same identifier, even
+if the data they hold is different, they are equal but not fungible.
+
+Examples:
+    >>> from openfisca_core import entities
+
+    >>> this = entities.Entity(1, "a", "b", "c")
+    >>> that = entities.Entity(1, "d", "f", "g")
+    >>> this == that
+    True
```

Comment:
This is not clear to me (but I am not an expert).
Could you elaborate more so I can understand what you are achieving with this lines of code ?

Reply:
Hi! The idea is that, as long as an entity and another entity have the same id, they're equal, even if they have different, let's say, documentation.

Reply:
How is it used ? Could you provide an example ?